rfcs/0024-incremental-ledger-catchup.md

```diff
@@ -0,0 +1,43 @@
+## Summary
+
+The origin transition catchup design has performance issues. In
+current implementation, we would always downloading the a list of transitions
+from requested hash to peer's root (plus the entire root history).
+This is not very efficient in a realistic setting where k is ~3000. The new
+design features
+1) avoids requesting unnecessary transitions;
+2) instead of having a monolithic request to download everything; try to
+   download small pieces from different peers at the same time.
+
+## Motivation
+
+The new design would improve the efficiency of doing ledger catchup.
+
+## Detailed design
+
+Split ledger catchup into 2 phases:
+
+1) instead of requesting a path/list of transitions from the requested hash to
+   peer's root (plus the root history), requesting a path/list of hashes. And
+   then use this path of hashes to compute the list of missing
+   transitions.
+   
+2) Depending on the size of the list of the missing transitions, spawn one or
+   more parallel requests to get the missing transitions from peers.
```

Comment:
what is the potential time saving here?

Reply:
I think the major issue with current implementation is download everything in one request. This RFC attempts to change it to download the thing piece by piece, and I think nothing prevents us to do it in parallel. But I think it would probably save some time or not depending on the network condition.

Reply:
It might be useful to try to quantify the potential time saving